hypershift/baremetal/none.md

```diff
@@ -0,0 +1,360 @@
+bash < <(curl -s -S -L https://raw.githubusercontent.com/moovweb/gvm/master/binscripts/gvm-installer)
+source ${HOME}/.gvm/scripts/gvm
+gvm install go1.17
+gvm use go1.17
+
+# build the binary
+make hypershift
+popd
+~~~
+
+Then, the `hypershift` binary can be moved to a convenient place as:
+
+~~~sh
+sudo install -m 0755 -o root -g root hypershift/bin/hypershift /usr/local/bin/hypershift
+~~~
+
+Alternatively, it can be compiled using a container as:
+
+~~~sh
+# Install prerequisites
+sudo dnf install podman -y
+# Compile hypershift
+mkdir -p ./tmp/ && \
+podman run -it -v ${PWD}/tmp:/var/tmp/hypershift-bin/:Z --rm docker.io/golang:1.17 sh -c \
+  'git clone --depth 1 https://github.com/openshift/hypershift.git /var/tmp/hypershift/ && \
+  cd /var/tmp/hypershift && \
+  make hypershift && \
+  cp bin/hypershift /var/tmp/hypershift-bin/'
+sudo install -m 0755 -o root -g root ./tmp/hypershift /usr/local/bin/hypershift
+~~~
+
+Once the binary is in place, the operator deployment is performed as:
+
+~~~sh
+hypershift install
+~~~
+
+Using `hypershift install --render > hypershift-install.yaml` will create a yaml file with all the assets required to deploy HyperShift. It has been [included in the assets folder](../assets/hypershift-install.yaml) as a reference.
+
+> **NOTE**: There are a few more flags to customize the HyperShift operator installation but they are out of the scope of this document.
+
+## **Deploy a hosted cluster**
+
+There are two main CRDs to describe a hosted cluster:
+* [`HostedCluster`](https://hypershift-docs.netlify.app/reference/api/#hypershift.openshift.io/v1alpha1.HostedCluster) defines the control plane hosted in the management OpenShift
+* [`NodePool`](https://hypershift-docs.netlify.app/reference/api/#hypershift.openshift.io/v1alpha1.NodePool) defines the nodes that will be created/attached to a hosted cluster
+
+The `hostedcluster.spec.platform` specifies the underlying infrastructure provider for the cluster and is used to configure platform specific behavior, so depending on the environment it is required to configure it properly.
+
+In this repo we will cover the 'none' provider.
+
+### **Deploy a 'none' hosted cluster and adding a bare metal worker**
+
+#### **Requisites**
+
+* DNS entry to point `api.${cluster}.${domain}` to each of the nodes where the hostedcluster will be running. This is because the hosted cluster API is exposed as a `nodeport`. For example:
+
+~~~sh
+api.hosted0.example.com.  IN A  10.19.138.32
+api.hosted0.example.com.  IN A  10.19.138.33
+api.hosted0.example.com.  IN A  10.19.138.37
+~~~
+
+* DNS entry to point `*.apps.${cluster}.${domain}` to a load balancer deployed to redirect incoming traffic to the ingresses pod (the (OpenShift documentation)[https://docs.openshift.com/container-platform/4.9/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-load-balancing-user-infra-example_installing-platform-agnostic] provides some instructions about this)
```

Comment:
(the (OpenShift documentation)[https://docs.openshift.com/container-platform/4.9/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-load-balancing-user-infra-example_installing-platform-agnostic] provides some instructions about this) 

The link is not being rendered properly, probably due to "(the"